docs/rules/no-api-reassignments.md

```diff
@@ -0,0 +1,53 @@
+# Prevent public property reassignments (no-api-reassignments)
+
+Reassignment public properties from with the component is generally a bad practice. The public properties are should be set by the parent components and should never be mutated. This linting rule prevents component to reassignment the properties they expose.
+
+## Rule details
+
+Example of **incorrect** code:
+
+```js
+import { LightningElement, api } from 'lwc';
+
+export default class Test extends LightningElement {
+    @api foo;
+
+    constructor() {
+        super();
+        this.foo = 1;
+    }
+
+    method() {
+        this.foo = 1;
+
+        return () => {
+            this.foo = 1;
+        };
+    }
+}
+```
+
+Example of **correct** code:
+
+```js
+export default class Test extends LightningElement {
+    @api foo = 1;
+
+    method() {
+        return function() {
+            this.foo = 1;
```

Comment:
I find this example very confusing.

Reply:
method should probably be turned into a public getter, oterwise this can never be invoked

Reply:
Agreed, this example adds more confusion than anything else. I wanted to illustrate that `this.foo` is allowed is the function expression scope because we don't know what is the `this` value. I will remove it.